core/src/main/resources/org/apache/spark/ui/static/stagepage.js

```diff
@@ -0,0 +1,926 @@
+                                    } else {
+                                        return "";
+                                    }
+                                } else {
+                                    return "";
+                                }
+                            },
+                            name: "Shuffle Spill (Memory)"
+                        },
+                        {
+                            data : function (row, type) {
+                                if ("taskMetrics" in row) {
+                                    if (row.taskMetrics.diskBytesSpilled > 0) {
+                                        return type === 'display' ? formatBytes(row.taskMetrics.diskBytesSpilled, type) : row.taskMetrics.diskBytesSpilled;
+                                    } else {
+                                        return "";
+                                    }
+                                } else {
+                                    return "";
+                                }
+                            },
+                            name: "Shuffle Spill (Disk)"
+                        },
+                        {
+                            data : function (row, type) {
+                                var msg = row.errorMessage;
+                                if (typeof msg === 'undefined'){
+                                    return "";
+                                } else {
+                                        var form_head = msg.substring(0, msg.indexOf("at"));
+                                        var form = "<span onclick=\"this.parentNode.querySelector('.stacktrace-details').classList.toggle('collapsed')\" class=\"expand-details\">+details</span>";
+                                        var form_msg = "<div class=\"stacktrace-details collapsed\"><pre>" + row.errorMessage + "</pre></div>";
+                                        return form_head + form + form_msg;
+                                }
+                            },
+                            name: "Errors"
+                        }
+                    ],
+                    "columnDefs": [
+                        { "visible": false, "targets": 11 },
+                        { "visible": false, "targets": 12 },
+                        { "visible": false, "targets": 13 },
+                        { "visible": false, "targets": 14 },
+                        { "visible": false, "targets": 15 },
+                        { "visible": false, "targets": 16 },
+                        { "visible": false, "targets": 17 },
+                        { "visible": false, "targets": 18 }
+                    ],
+                };
+                var taskTableSelector = $(taskTable).DataTable(task_conf);
+                $('#active-tasks-table_filter input').unbind();
+                $('#active-tasks-table_filter input').bind('keyup', function(e) {
+                  taskTableSelector.search( this.value ).draw();
```

Comment:
search on the task table seems to do query upon typing, not sure that is what we want, would be better if use completed and hit enter since its doing server side query

Reply:
Implemented search on keyup with timeout, thank you.